src/libsystemd-network/dhcp-option.c

```diff
@@ -93,32 +112,74 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
 
                 options[*offset] = code;
                 options[*offset + 1] = l;
-
                 *offset += 2;
 
                 ORDERED_SET_FOREACH(p, s) {
-                        options[*offset] = p->option;
-                        options[*offset + 1] = p->length;
-                        memcpy(&options[*offset + 2], p->data, p->length);
-                        *offset += 2 + p->length;
+                        assert_se(dhcp_option_append_tlv(options, size, offset, p->option, p->length, p->data) >= 0);
+                }
+                break;
+        }
+        case SD_DHCP_OPTION_RELAY_AGENT_INFORMATION: {
+                sd_dhcp_server *server = (sd_dhcp_server *) optval;
+                size_t current_offset = *offset + 2;
```

Comment:
The offset value must be checked before calling `dhcp_option_append_tlv()` below.

Reply:
Hi! Not sure I understood. Is it the same as in https://github.com/systemd/systemd/pull/19384#discussion_r623675293 ?

Reply:
Hmm?? The location of this comment seems shifted while I was writing it, as you committed during I review. I will re-post the comment after reviewing again if the issue still exists.